File: Pharmacy-management-System.-master/Angular/PharmacyProject/src/app/shared/services/drug.service.ts
import { Drug } from './../models/drug';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class DrugService {
  readonly rootUrl = 'https://localhost:44348';
  constructor(private http: HttpClient) {}

  GetDrugs(): Observable<Drug[]> {
    return this.http.get<Drug[]>(this.rootUrl + '/api/Drug/GetAllDrugs');
  }

  AddDrugs(drug: Drug): Observable<Drug> {
    return this.http.post<Drug>(this.rootUrl + '/api/Drug/AddDrug', drug);
  }

  DeleteDrug(id: number): Observable<Drug> {
    return this.http.delete<Drug>(this.rootUrl + '/api/Drug/DeleteDrug/' + id);
  }

  UpdateDrug(id: number, drug: Drug): Observable<Drug> {
    return this.http.put<Drug>(this.rootUrl + '/api/Drug/EditDrug/' + id, drug);
  }

  GetDrugById(id: number): Observable<Drug> {
    return this.http.get<Drug>(this.rootUrl + '/api/Drug/SearchDrug/' + id);
  }
}

File: Pharmacy-management-System.-master/Pharmacy_Management_System/Services/DrugService.cs
using Pharmacy_Management_System.Model;
using Pharmacy_Management_System.Repository;
using System.Collections.Generic;

namespace Pharmacy_Management_System.Services
{
    public class DrugService
    {
        private readonly DrugDAL _drug;
        public DrugService(DrugDAL drug)
        {
            _drug = drug;
        }

        public string AddDrug(Drug drugs)
        {
            return _drug.AddDrug(drugs);

        }

        public void DeleteDrug(int id)
        {
            _drug.DeleteDrug(id);
        }

        public List<Drug> GetAllDrugs()
        {
            return _drug.GetAllDrugs();
        }

        public Drug GetDrugById(int id)
        {
            return _drug.GetDrugById(id);
        }

        public void UpdateDrug(int id, Drug drug)
        {
            _drug.UpdateDrug(id, drug);
        }
    }
}
